### DISCUSSION TITLE
watch height of element and store value in store

### DISCUSSION THREAD

**Discussion by StefanVesper** [alpinejs/alpine#2998] [1. Help] [answered]
Hi there! 

Is it possible to watch the height of an element and store the value "reactively" in a store? 

I want to have a reference to the height of a header element (it is changing on scroll) and pass it to some non-alpine Javascript.

Any ideas to acomplish that? 

Thank you ! 

Steve

**Comment by danddanddand** [comment, marked as answer]
Something like this... 

`<div x-data @scroll.window="$store.blah.yourValue = $el.getBoundingClientRect().height"> ... </div>
`

You may want to debounce the scroll a bit depending on what you're doing.
